content/news/04-clash122.md

```diff
@@ -0,0 +1,68 @@
+
+```
+a `seq` b
+```
+
+you'd write the following instead:
+
+```
+a `seqX` b
+```
+
+Great!
+
+A very typical pattern in memory primitives is to (indirectly) use these functions using [defaultSeqX](https://hackage.haskell.org/package/clash-prelude-1.2.2/docs/Clash-XException.html#v:defaultSeqX) like:
+
+```
+a `defaultSeqX` a :- as
+```
+
+The purpose of this expression is to force a (WH)NF evaluation of `a` before being able to observe the _spine_ of the signal. I.e., evaluate a value at time _t_ if you want to observe the value at _t+1_. This solves the same problem as the one described under [building up unevaluated expressions](https://wiki.haskell.org/Memory_leak) at the Haskell wiki. It turns out that this expression was parsed as:
+
+```
+(a `defaultSeqX` a) :- as
+```
+
+instead of what we actually wanted and expected:
+
+```
+a `defaultSeqX` (a :- as)
+```
+
+We've since added ```infixr 0 `defaultSeqX` ```. The expression now parses as expected and we've observed that this fixes the memory issues our clients were seeing! It turns out that `deepseq` has exactly the same problem, for which we've [submitted a patch](https://github.com/haskell/deepseq/pull/56).
+
+In short, huge thanks to [Dr. Gergő Érdi](https://github.com/gergoerdi) who pushed us to get to the bottom of the issue and release a fix for it 🎉.
+
+# Plans for 1.4
+The next major relase of Clash will be 1.4. Two major additions are in the pipeline:
+
+1. _Ground work for a Partial Evaluator._ Clash currently uses a transformation based rewrite system to translate Haskell into synthesizable circuits. This transformation system was [originally written](https://research.utwente.nl/en/publications/digital-circuit-in-c%CE%BBash-functional-specifications-and-type-direc) to make all terms in GHC's/Clash's Core IR completely monomorphic and first-order. Bit by bit, more and more transformations were added: removing case statements based on a known subject, unrolling primitive definitions, removing lambdas by propagating their arguments, and many more. As the number of transformations grew, so did the complexity and predictability. In fact, we're currently seeing issues that are very hard to solve within a transformation based system.
```

Comment:
```suggestion
1. _Ground work for a Partial Evaluator._ Clash currently uses a transformation based rewrite system to translate Haskell into synthesizable circuits. This transformation system was [originally written](https://research.utwente.nl/en/publications/digital-circuit-in-c%CE%BBash-functional-specifications-and-type-direc) to make all terms in GHC's/Clash's Core IR completely monomorphic and first-order. Bit by bit, more and more transformations were added: removing case statements based on a known subject, unrolling primitive definitions, removing lambdas by propagating their arguments, and many more. As the number of transformations grew, so did the complexity and unpredictability. In fact, we're currently seeing issues that are very hard to solve within a transformation based system.
```

Reply:
It appears this copyedit suggestion I made got lost somewhere in the heap of small changes :-). I suggested this change: "complexity and predictability grew" -> unpredictability

Reply:
oh oops, sorry

Reply:
I don't think the GitHub review system works very well on Markdown... it's more for < 80 columns of code, rather than full paragraphs... so I blame the system :-)

Reply:
Fixed!